src/generator/python/index.ts

```diff
@@ -90,58 +90,104 @@ export class PyGenerator implements CodeGenerator {
       parts.push(`        self.${field.name} = ${field.name}`);
     }
 
-    // encode method
-    const encodeParts: string[] = [];
-    encodeParts.push("");
-    encodeParts.push("    def encode(self, sia: Sia) -> Sia:");
+    return [
+      parts.join("\n"),
+      this.encodeMethod(schema),
+      this.decodeMethod(schema),
+    ].join("\n");
+  }
+
+  private encodeMethod(schema: SchemaDefinition): string {
+    const lines: string[] = [];
+    lines.push("");
+    lines.push("    def encode(self, sia: Sia) -> Sia:");
+
     for (const field of schema.fields) {
-      const fn = this.getSerializeFunctionName(field);
       const valueExpr = `self.${field.name}`;
-      const isAscii = field.type == "string8" && field.encoding == "ascii";
-      const comment = isAscii ? "  # ascii" : "";
-
-      let call: string;
-      if (fn.startsWith("sia.")) {
-        call = `${fn}(${valueExpr})`;
-      } else if (this.knownSchemas.has(field.type)) {
-        call = `${valueExpr}.encode(sia)`;
+
+      if (field.isArray) {
+        const serializeElemFn = this.getSerializeFunctionName(field);
+        const serializeElemArgs = this.getSerializeFunctionArgs(field);
+        let lambda: string;
```

Comment:
Don't use let here. Make a private method and call that:

`const lambda = this.appropriatefunctionname(params)`